Amazon.IonObjectMapper/IonSerializer.cs

```diff
@@ -327,6 +333,15 @@ public void Serialize<T>(IIonWriter writer, T item)
 
             if (item is object)
             {
+                var customSerializerAttribute = item.GetType().GetCustomAttribute<IonSerializerAttribute>();

```

Comment:
Would it be better to have this logic defined inside `IonObjectSerializer:Serialize()` instead of `IonSerializer.cs` (and likewise for `Deserialize()`)?

Any opinion on this @alpian ?

Reply:
Personally I think if we have this logic defined in `IonSerializer.cs`, if the `item` has `IonSerializerAttribute`, we can directly create `customSerializer` rather than creating `IonObjectSerializer` then `customSerializer`.

Defined in `IonObjectSerializer:Serialize()` will definitely make the logic in `IonSerializer.cs` clearer though.

Reply:
It's an interesting idea having the object serializer delegate to another one. I'm ok with it being here for now.